### PR TITLE
Add TF contrastive image text finetuning example

### PR DESCRIPTION
**Pull request by Rocketknight1**
This PR adds a TF port of the PyTorch example for finetuning the `TFVisionTextDualEncoderModel` class. Functionality is largely the same, but I used a `tf.data` pipeline to efficiently stream images instead of `torchvision`. I also added the ability to specify separate image/text models with arguments, whereas in the PyTorch example you have to create the dual encoder with a separate script.

I also caught a small bug in the original model code while writing this - loss is a scalar rather than having shape `(1,)`. That's fixed in here too!

### CODE REVIEW

**Review by sgugger — APPROVED** [review]
Thanks for adding this!

**Comment by Rocketknight1** [comment]
Update: Woah! The argument names to `PushToHubCallback` across our examples became outdated and tests weren't picking that up. I'm fixing it in this PR too.

File: examples/tensorflow/contrastive-image-text/run_clip.py [hunk 1]
```diff
@@ -0,0 +1,587 @@
...
+    logger.setLevel(log_level)
+    transformers.utils.logging.set_verbosity(log_level)
+    transformers.utils.logging.enable_default_handler()
+    transformers.utils.logging.enable_explicit_format()
+
+    # Log on each process the small summary:
+    logger.warning(
+        f"Process rank: {training_args.local_rank}, device: {training_args.device}, n_gpu: {training_args.n_gpu}"
+        + f"distributed training: {bool(training_args.local_rank != -1)}, 16-bits training: {training_args.fp16}"
+    )
```
Comment (gante) [inline examples/tensorflow/contrastive-image-text/run_clip.py:290]: I think `local_rank` is a `torch.distributed`-only thing 🤔

**Review by gante — APPROVED** [review]
LGTM 👍  Thanks for adding the example 💛

File: examples/tensorflow/contrastive-image-text/run_clip.py (hunk 1 above)
Comment (Rocketknight1) [reply examples/tensorflow/contrastive-image-text/run_clip.py:290]: Oops, good catch! I thought I'd gotten rid of all of the torch stuff.
